From 113978e7339cbb51dc1fcb2908543249e8e6856e Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Wed, 7 Aug 2024 05:41:37 +0000
Subject: [PATCH] [wasm] `strlcpy` and `strlcat` are available in wasi-libc

Mark them available on WASI. Otherwise, the `static inline` implementations
are activated and the build fails with multiple definitions.
---
 .../CoreFoundation/internalInclude/CoreFoundation_Prefix.h   | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Sources/CoreFoundation/internalInclude/CoreFoundation_Prefix.h b/Sources/CoreFoundation/internalInclude/CoreFoundation_Prefix.h
index 9ef8f64a..dea3b575 100644
--- a/Sources/CoreFoundation/internalInclude/CoreFoundation_Prefix.h
+++ b/Sources/CoreFoundation/internalInclude/CoreFoundation_Prefix.h
@@ -109,6 +109,11 @@ typedef char * Class;
 #include <pthread.h>
 #endif
 
+#if TARGET_OS_WASI
+#define HAVE_STRLCPY 1
+#define HAVE_STRLCAT 1
+#endif
+
 #if TARGET_OS_WIN32
 #define BOOL WINDOWS_BOOL
 
-- 
2.43.2

